salvo/src/lib/api/source.proto

```diff
@@ -0,0 +1,35 @@
+syntax = "proto3";
+
+package salvo;
+
+// Capture the location of sources needed for the benchmark
+message SourceRepository {
+  // Specify whether this source location is Envoy or NightHawk
+  oneof identity {
+    bool envoy = 1;
+    bool nighthawk = 2;
+  }
+
+  // Specify the location of the source repository on disk. If specified
+  // this location is used to determine the origin url, branch, and commit
+  // hash. If not specified, the remaining fields must be populated
```

Comment:
Seems this would be a good application for a oneof. That way we can make the API self-explanatory when it comes to understanding which fields should be populated and which are mutually exclusive.

We can add two separate messages, e.g. LocalSourceLocation and RemoteSourceLocation. The LocalSourceLocation would only contain a single string field `directory_path`. The RemoteSourceLocation can contain all the otherfields. The parent message SourceRepository can then have a oneof between the two messages. This will also simplify a lot of the comments here, since we no longer need to say things like "This is ignored if...".

Reply:
Updated